packages/frontend/editor-ui/src/app/components/WorkflowProductionChecklist.vue

```diff
@@ -135,17 +142,62 @@ const availableActions = computed(() => {
 		});
 	}
 
-	if (isMcpAvailable.value && !suggestedActionSettings['mcp-access']?.ignored) {
-		actions.push({
-			id: 'mcp-access',
-			title: i18n.baseText('mcp.productionCheklist.title'),
-			description: i18n.baseText('mcp.productionCheklist.description'),
-			moreInfoLink: MCP_DOCS_PAGE_URL,
-			completed: props.workflow.settings?.availableInMCP ?? false,
-		});
+	// MCP access action
+	const mcpAction = getMcpAction();
+	if (mcpAction) {
+		actions.push(mcpAction);
 	}
 
 	return actions;
+
+	function getMcpAction(): {
+		id: ActionType;
+		title: string;
+		description: string;
+		moreInfoLink: string;
+		completed: boolean;
+	} | null {
+		if (!isMcpModuleEnabled.value) return null;
+
+		const baseAction = {
+			title: i18n.baseText('mcp.productionChecklist.title'),
+			moreInfoLink: MCP_DOCS_PAGE_URL,
+		};
+
+		// Instance-level MCP access is disabled - show action to enable it
+		if (!isMcpAccessEnabled.value) {
+			// Only show to admins if not ignored
+			if (
+				!canToggleInstanceMCPAccess.value ||
+				suggestedActionSettings['instance-mcp-access']?.ignored ||
+				!isWorkflowEligibleForMcpAccess.value
+			) {
+				return null;
+			}
+
+			return {
+				...baseAction,
+				id: 'instance-mcp-access',
+				description: i18n.baseText('mcp.productionChecklist.instance.description'),
+				completed: false,
+			};
+		}
+
+		// Workflow-level MCP access (instance-level is enabled)
+		if (
+			!isWorkflowEligibleForMcpAccess.value ||
```

Comment:
Should we move this check `!isWorkflowEligibleForMcpAccess.value` at the top? Seems like we always return null if the result is true?